src/gl_fns.rs

```diff
@@ -1360,10 +1361,10 @@ impl Gl for GlFns {
         }
     }
 
-    fn get_vertex_attrib_iv(&self, index: GLuint, pname: GLenum) -> GLint {
-        let mut result = 0 as GLint;
+    fn get_vertex_attrib_iv(&self, index: GLuint, pname: GLenum) -> Vec<GLint> {
```

Comment:
This always returns a vector of length 4, we may as well return a `[GLint; 4]` in that case, right?

Reply:
It's 4 values only when `pname` is `GL_CURRENT_VERTEX_ATTRIB`, other cases return just 1 value.

Reply:
What do you think about adding a little optimization like so?
```Rust
let mut result = if pname == ffi::CURRENT_VERTEX_ATTRIB {
    vec![0 as GLint; 4]
} else {
    vec![0 as GLint; 1]
}
```

Reply:
It would be nice to avoid excessive heap allocations. We are already doing it in quite a few places, but here at least it doesn't feel as necessary.

Reply:
Updated the PR and added this little optimization.

Reply:
What I've decided to do for other APIs that return sequences and whatnot is that `get_vertex_attrib_iv` should just not exist, and we should instead provide a `current_vertex_attrib` method. See `get_viewport` for example.